### PR TITLE
Update pyarpack tests with SPD solvers

### PR DESCRIPTION
**Pull request by FabienPean**
My proposal for fixing #432 

* Swapping matrices `pyarpackSparseLDLT` &lrarr; `pyarpackSparseLU`
* Swapping matrices `pyarpackDenseLDLT` &lrarr; `pyarpackDenseLUPP`
* `pyarpackCGILU` solves for hermitian matrices (and only non-general due to risk of losing positive definitness with shift-inverse)

### CODE REVIEW

File: EXAMPLES/PYARPACK/pyarpackSparseLU.py.in [hunk 1]
```diff
@@ -113,4 +122,4 @@ for val, vec in zip(arpackSlv.val, arpackSlv.vec):
     print("eigen value:", val)
     print("eigen vector:")
     for v in range(n):
-        print(vec[v])
+        print(vec[v])
```
Comment (fghoussen) [inline EXAMPLES/PYARPACK/pyarpackSparseLU.py.in:125]: I guess this is no op! Can you remove it and update the changelog? Thanks!

**Comment by fghoussen** [comment]
OK, make sense too! :)

File: EXAMPLES/PYARPACK/pyarpackSparseLU.py.in (hunk 1 above)
Comment (FabienPean) [reply EXAMPLES/PYARPACK/pyarpackSparseLU.py.in]: Did you mean restoring the EOL ? If so I restored it in the newer commit. Also fixed the solver parameters to match their respective solver (slvOffset and slvScale moved back to SparseLDLT, slvPvtThd to SParseLU, etc)

Comment (fghoussen) [reply EXAMPLES/PYARPACK/pyarpackSparseLU.py.in]: thanks

**Pull request merged**